backend/tarsy/services/history_service.py

```
@@ -643,14 +599,23 @@ def get_sessions_list(
                 interaction_counts = result.get('interaction_counts', {})
                 total_items = result.get('pagination', {}).get('total_items', 0)
                 
-                # Add interaction counts as dynamic attributes to session objects
-                for session in sessions:
-                    counts = interaction_counts.get(session.session_id, {})
-                    # Use object.__setattr__ to bypass SQLModel validation
-                    object.__setattr__(session, 'llm_interaction_count', counts.get('llm_interactions', 0))
-                    object.__setattr__(session, 'mcp_communication_count', counts.get('mcp_communications', 0))
+                # Handle sessions as dicts (repository now returns dicts for backward compatibility)
```

Ruff (0.12.2)

Line too long (100 > 88)

(E501)